doc: doc for new 'atlas migrate lint' behavior (#2072)
ariga/atlas: doc/md/lint/analyzers.mdx

Below are the `Analyzer` implementations currently supported by Atlas.

### Non-Linear Changes

Non-additive changes, often referred to as non-linear changes, are changes to the migration directory that are not added

> in a sequential order. This is a bit like the linear history in version control systems where the migration files are

Reviewer: ```suggestion
in a sequential order. This is a bit like the linear history in version control systems, where migration files are
```

commits and the migration directory is the repository. Let's explain with three examples why ensuring the linearity of the
migration directory is important:

> - When a developer introduce a new migration file without having the **latest state** of the directory, there is a risk

Reviewer: ```suggestion
- When a developer introduces a new migration file without having the **latest state** of the directory, there is a risk
```

>   of generating an incorrect file that might be conflicted with the actual schema and cause a failure during the deployment.

Reviewer: ```suggestion
  of generating an incorrect file that might conflict with the actual schema and cause a failure during deployment.
```

- When a developer merges a feature branch to the main branch with a new migration file that is not positioned at the

>   end of the directory, there is a risk that Atlas will skip this migration file as there might be some databases that already

Reviewer: ```suggestion
  end of the directory, there is a risk that Atlas will skip this migration file, as there might be some database that already
```

>   contains a higher version of the migration directory. This scenario can cause an unexpected and surprising behavior during

Reviewer: ```suggestion
  contains a higher version of the migration directory. This scenario can cause unexpected and surprising behavior during
```
